skore/src/skore/project/metadata.py

```diff
@@ -37,6 +38,116 @@ class Metadata(DataFrame):
 
     _metadata = ["project"]
 
+    @staticmethod
+    def _iterate_in_pipe_tree(step, itemized_pipeline: list, depth: int) -> list:
+        """
+        Iterate through the pipeline tree.
+
+        Parameters
+        ----------
+        step : Pipeline, FeatureUnion or TableVectorizer
+            The step to explore.
+
+        Returns
+        -------
+        list
+            A list of estimators or transformers in the pipeline.
+        """
+        type_ = type(step)
+        if type_ == Pipeline:
+            itemized_pipeline = Metadata._explore_sk_pipeline(
+                step, itemized_pipeline, depth=depth + 1
+            )
+        elif type_ == FeatureUnion:
+            itemized_pipeline = Metadata._explore_sk_feature_union(
+                step, itemized_pipeline, depth=depth + 1
+            )
+        elif str(type_) == "<class 'skrub._table_vectorizer.TableVectorizer'>":
```

Comment:
I'm not sure it's how it should be done, but I see no other way without importing skrub, which we don't want to have mandatory in our imports.